react/admin/CostCenterDetails.tsx

```diff
@@ -143,23 +148,54 @@ const CostCenterDetails: FunctionComponent = () => {
     }))
   }
 
+  const handleSetAddresses = (_addresses: Address[]) => {
+    if (!_addresses.find(item => item.checked))
+      setAddresses(
+        _addresses.map((item, index) => {
+          item.checked = index === 0
+
+          return item
+        })
+      )
+  }
+
+  const handleCheckDefault = (address: Address) => {
+    setAddresses(
+      addresses.map(item => {
+        item.checked = item === address
+
+        return item
+      })
+    )
+  }
+
   useEffect(() => {
+    if (data?.getCostCenterById?.addresses)
+      handleSetAddresses(data.getCostCenterById.addresses)
+
     if (addresses.length || !data?.getCostCenterById?.addresses?.length) return
```

Comment:
I think `handleSetAddresses` should happen after the `if (address.length || ...` conditional. I think the current code is causing a bug. You can reproduce the bug like so:

- Go to `/admin/b2b-organizations/organizations`
- Click into an organization
- Click into a cost center
- Click the "back to organization" link
- Click into the same cost center again
- The list of addresses will be blank

Reply:
Yes, I got it too.
It's done